### PR TITLE
#536 Add config for membership cancellation notice period

### LINKED ISSUE #536: Add configuration for membership cancellation notice period
As far as I can see, we currently don't have any configuration for the notice period (number of month before next cancellation date) for the membership. So members can cancel their membership a day before the end of the `MEMBERSHIP_END_MONTH` although most coops might have several month of notice period stated in their conditions. There is even a wrong comparison to the subscription annual cancellation month `BUSINESS_YEAR_CANCELATION_MONTH`.

Proposal: Add configuration option `MEMBERSHIP_END_NOTICE_PERIOD` in months and adapt logic accordingly.

### PR DESCRIPTION
**Pull request by nixnuex**
Closes #536

### CODE REVIEW

**Comment by nixnuex** [comment]
Also closes #538

File: juntagrico/templates/manage_shares.html [hunk 1]
```diff
@@ -63,11 +63,13 @@ <h4>{% trans "Übersicht" %}</h4>
...
-                                {% elif  share.cancelled_date is None and share.paid_date is None %}
+                                {% elif share.cancelled_date is None and share.paid_date is None %}
                                     {% blocktrans %}Diese/s/r {{ v_share }} wurde noch nicht als bezahlt verbucht{% endblocktrans %}
+                                {% elif ibanempty %}
+                                    {% blocktrans %}Kann gekündigt werden, sobald eine IBAN hinterlegt ist{% endblocktrans %}
                                 {% elif  share.cancelled_date is None and share.paid_date is not None %}
                                     <a href="{% url 'share-cancel' share.id %}"class="btn btn-success"
-                                       onclick="return confirm('{% blocktrans %} Möchtest du sicher diese/s/n {{ v_share }} verbindlich kündigen? {% endblocktrans%}')">
+                                       onclick="return confirm('{% blocktrans %} Möchtest du sicher diese/s/n {{ v_share }} verbindlich zum {{ canceldate }} kündigen? {% endblocktrans%}')">
```
Comment (pascalfree) [inline juntagrico/templates/manage_shares.html:72]: Using the chance to fix the sentence
```suggestion
                                       onclick="return confirm('{% blocktrans %} Möchtest du diese/s/n {{ v_share }} verbindlich zum {{ canceldate }} kündigen? {% endblocktrans%}')">
```

File: juntagrico/views_subscription.py [hunk 2]
```diff
@@ -426,6 +426,8 @@ def manage_shares(request):
         'shares': shares.all(),
         'shareerror': shareerror,
         'required': member.required_shares_count,
+        'ibanempty': not member.iban,
+        'canceldate': next_membership_end_date(),
```
Comment (pascalfree) [inline juntagrico/views_subscription.py:430]: call it 'next_membership_end_date'.

File: juntagrico/templates/manage_shares.html [hunk 3]
```diff
@@ -63,11 +63,13 @@ <h4>{% trans "Übersicht" %}</h4>
                                     {% blocktrans %}Du brauchst mindestens 1 {{ v_share }}{% endblocktrans %}
                                 {% elif forloop.counter <=  required %}
                                     {% blocktrans %}Diese/s/r {{ v_share }} wird in einem/r {{ v_subscription }} benötigt{% endblocktrans %}
-                                {% elif  share.cancelled_date is None and share.paid_date is None %}
+                                {% elif share.cancelled_date is None and share.paid_date is None %}
                                     {% blocktrans %}Diese/s/r {{ v_share }} wurde noch nicht als bezahlt verbucht{% endblocktrans %}
+                                {% elif ibanempty %}
+                                    {% blocktrans %}Kann gekündigt werden, sobald eine IBAN hinterlegt ist{% endblocktrans %}
```
Comment (pascalfree) [inline juntagrico/templates/manage_shares.html:69]: Add a link here to the form where the IBAN can be set.

**Review by pascalfree — CHANGES_REQUESTED** [review]

**Review by pascalfree — APPROVED** [review]

**Pull request merged**